src/banyan_wasm/compat/types/node_metadata.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{banyan_wasm::TombWasmError, value};
+use gloo::console::log;
+use js_sys::{Object, Reflect};
+use std::collections::BTreeMap;
+use wasm_bindgen::prelude::*;
+use wnfs::{common::Metadata as NodeMetadata, libipld::Ipld};
+
+/// Wrapper around a NodeMetadata
+#[derive(Debug)]
+pub struct WasmNodeMetadata(pub(crate) NodeMetadata);
+
+impl TryFrom<JsValue> for WasmNodeMetadata {
+    type Error = TombWasmError;
+
+    fn try_from(js_value: JsValue) -> Result<Self, Self::Error> {
+        let object = js_value
+            .dyn_into::<Object>()
+            .map_err(|_| TombWasmError("expected an object to be passed in".to_string()))?;
+
+        let mut map = BTreeMap::new();
+
+        // We know object is an Object already, so this shouldn't be able to panic (that is the
+        // only documented way for this to throw an error).
+        let created_ref =
+            Reflect::get(&object, &JsValue::from_str("created")).expect("undocumented error");
+        if let Some(timestamp) = created_ref.as_f64() {
+            map.insert("created".into(), Ipld::Integer(timestamp as i128));
+        } else {
+            log!("WARNING: WasmNodeMetadata did not contain a 'created' timestamp");
+        }
+
+        // See created
+        let modified_ref =
```

Comment:
Again, these will cause frontend errors, its better to panic

Reply:
not part of this PR, just moved code